torchx/lib/torchx/backend.ex

```diff
@@ -150,6 +150,44 @@ defmodule Torchx.Backend do
     NIF.transpose(to_ref(t), 0, 1) |> from_ref(out)
   end
 
+  @impl true
+  def slice(%T{shape: shape} = out, %T{} = t, start_indices, lengths, strides) do
+    ref = to_ref(t)
+
+    for dim <- 0..(length(start_indices) - 1), reduce: ref do
+      ref -> NIF.narrow(ref, dim, Enum.at(start_indices, dim), Enum.at(lengths, dim)) |> unwrap!()
+    end
```

Comment:
Don't use `0..(length(start_indices) - 1)` because it will be wrong for scalars (tensors with no dimensions) as the range will have one element but there is no actual dimension. Prefer to use `Nx.axes(shape)` instead.

This code can also be more efficient if you avoid narrowing when you know there is nothing to narrow. Here is how I would implement it:

    t
    |> to_ref()
    |> narrow(start_indices, lengths, 0, shape)

And then:

    defp narrow(ref, [start | starts], [length | lengths], axis, shape) do
      dim = elem(shape, axis)

      # Nothing to narrow
      if start == 0 and length == dim do
        narrow(ref, starts, lengths, axis + 1, shape)
      else
        ref
        |> NIF.narrow(ref, axis, start, length)
        |> narrow(starts, lengths, axis + 1, shape)
      end
    end

    defp narrow(ref, [], [], _axis, _shape), do: ref

I haven't tested it. :)